app/locations/middleware.js

```diff
@@ -17,20 +17,27 @@ function checkLocationsLength(req, res, next) {
 
 function setLocation(req, res, next) {
   const { locationId } = req.params
-  const { isRegion } = req.query
-
-  if (isRegion) {
-    req.session.currentLocation = null
-    req.session.currentRegion = find(req.session.regions, { id: locationId })
-    return next()
-  }
 
   const location = find(req.userLocations, { id: locationId })
 
   req.session.currentLocation = location
   next()
 }
 
+function setRegion(req, res, next) {
+  const { regionId } = req.params
+
+  req.session.currentLocation = null
```

Comment:
Is there any reason we set `currentLocation` to `null` but `currentRegion` to `undefined` when they're not set?

Reply:
`find` returns `undefined` when the item isn't found in the list. I thought I'd keep the same return value. I can change it to `null`.

Reply:
I think `undefined` makes sense. Maybe we set `req.session.currentLocation` to undefined too for consistency?

Reply:
I removed the `getRegions` and `find` and replaced it with a `getRegionById`.